products/views.py

```diff
@@ -1,3 +1,62 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import Q, Sum
+
+from products.models import Product
+
+class ProductsView(View):
+    def get(self,request):
+        category        = request.GET.get('category',None)
+        sub_category    = request.GET.get('sub_category',None)
+        detail_category = request.GET.get('detail_category',None)
+        color           = request.GET.getlist('color',None)
+        sorting         = request.GET.get('sorting',None)
+        search          = request.GET.get('KeyWord',None)
+        limit           = request.GET.get('limit',None)
+        offset          = int(request.GET.get('offset',0))
+    
+        filters = Q()
+
+        if category :
+            filters &= Q(detail_category__sub_category__category= category)
+
+        if sub_category :
+            filters &= Q(detail_category__sub_category__option = sub_category)
+
+        if detail_category :
+            filters &= Q(detail_category__detail = detail_category)
+        
+        if color :
+            filters &= Q(productoption__color__in = color)
+
+        if search :
+            filters = (Q(name__icontains = search)|
+                Q(detail_category__detail__icontains = search)|
+                Q(detail_category__sub_category__option__icontains = search)|
+                Q(detail_category__sub_category__category__name__icontains = search)
+                )
+        
+        sort_name = {
+            'new'        : '-create_at',
+            'bestseller' : '-sales',
+        }
+
+        if limit:
+            limit = offset + int(limit)
+
+        products = Product.objects.filter(filters).annotate(sales = Sum('productoption__sales'))\
+            .order_by(sort_name.get(sorting,'id'))[offset:limit]
+
+        results = [{
+            'id'         : product.id,
+            'name'       : product.name,
+            'price'      : round(product.price),
+            'main_image' : str(product.main_image),
+            'sales'      : product.sales,
+            'detail'     : product.detail_category.detail,
+            'color'      : [product_option.color.name for product_option in product.productoption_set.all()][0],
```

Comment:
```suggestion
            'color'      : [product_option.color.name for product_option in product.productoption_set.all()][0],
```
index을 상요한 이유를 코멘트로 남셔주세요.

Reply:
product_option 테이블에 size_id와 color_id 컬럼이 있는데 size는 3개 이고 color는 1개로 되어있는 구조라서 결과값에 동일한 color 3개가 표현되서 하나의 값만 가져오기위해 사용했습니다.

Reply:
이 부분은 한 번 같이 볼게요!!